chpl/chpl-service/src/main/java/gov/healthit/chpl/criteriaattribute/CriteriaAttributeValidator.java

```diff
@@ -0,0 +1,227 @@
+            messages.add(errorMessageUtil.getMessage("criteriaAttribute.edit.duplicate", context.getName()));
+        }
+
+        if (context.getRuleRequired()
+                && context.getCriteriaAttribe().getRule() == null) {
+            messages.add(errorMessageUtil.getMessage("criteriaAttribute.edit.emptyRule"));
+        }
+
+        if (context.getCriteriaAttribe().getRule() != null
+                && ruleDAO.getRuleEntityById(context.getCriteriaAttribe().getRule().getId()) == null) {
+            messages.add(errorMessageUtil.getMessage("criteriaAttribute.edit.notFoundRule"));
+        }
+
+        if (messages.size() > 0) {
+            ValidationException e = new ValidationException(messages);
+            throw e;
+        }
+
+    }
+
+    public void validateForDelete(CriteriaAttributeValidationContext context) throws ValidationException {
+        List<CertifiedProductDetailsDTO> listings = new ArrayList<CertifiedProductDetailsDTO>();
+        try {
+            listings = context.getCriteriaAttributeDAO().getCertifiedProductsByCriteriaAttribute(context.getCriteriaAttribe());
+        } catch (EntityRetrievalException ex) {
+            throw new ValidationException(ex.getMessage());
+        }
+
+        if (!CollectionUtils.isEmpty(listings)) {
+            String message = errorMessageUtil.getMessage("criteriaAttribute.delete.listingsExist",
+                    context.getName(),
+                    listings.size(),
+                    listings.size() > 1 ? "s" : "",
+                    listings.stream()
+                            .map(listing -> listing.getChplProductNumber())
+                            .collect(Collectors.joining(", ")));
+            ValidationException e = new ValidationException(message);
+            throw e;
+        }
+    }
+
+    private Set<String> validateCriteriaRemovedFromCriteriaAttribute(CriteriaAttributeValidationContext context) {
+        Set<String> messages = new HashSet<String>();
+        CriteriaAttribute origCriteriaAttribute = context.getCriteriaAttributeDAO().getCriteriaAttributeById(context.getCriteriaAttribe().getId());
+
+        getCriteriaRemovedFromCriteriaAttribute(context.getCriteriaAttribe(), origCriteriaAttribute).stream()
+                .forEach(crit -> {
+                    List<CertifiedProductDetailsDTO> listings = new ArrayList<CertifiedProductDetailsDTO>();
+                    try {
+                        listings = context.getCriteriaAttributeDAO().getCertifiedProductsByCriteriaAttributeAndCriteria(origCriteriaAttribute, crit);
+                    } catch (EntityRetrievalException ex) {
+                        messages.add(ex.getMessage());
+                    }
+
+                    if (!CollectionUtils.isEmpty(listings)) {
+                        messages.add(errorMessageUtil.getMessage("criteriaAttribute.edit.deletedCriteria.listingsExist",
+                                CertificationCriterionService.formatCriteriaNumber(crit),
```

Comment:
I think you might want to do the same thing here as above where we don't include the listings at the end of error message if there is more than some set amount. Check SVAP code to see what is currently done there.